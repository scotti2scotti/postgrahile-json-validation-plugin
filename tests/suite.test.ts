import * as Mutations from './fixtures/graphql'
import { close, query, start } from './helpers/server'

describe('JsonValidationPlugin', () => {
  beforeAll(async done => {
    await start()
    done()
  })
  afterAll(done => {
    close()
    done()
  })

  Object.entries(Mutations).forEach(([name, code]) => {
    test(name, async () => {
      expect.assertions(1)
      const result = await query(code)
      expect(result).toMatchSnapshot()
    })
  })
})
