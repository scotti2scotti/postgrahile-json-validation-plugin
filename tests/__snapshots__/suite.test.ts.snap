// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonValidationPlugin CreateMultipleInvalid 1`] = `
Object {
  "data": Object {
    "createDocument": null,
  },
  "errors": Array [
    Object {
      "locations": Array [
        Object {
          "column": 3,
          "line": 3,
        },
      ],
      "message": "foo should be integer, bar should be string",
      "path": Array [
        "createDocument",
      ],
    },
  ],
}
`;

exports[`JsonValidationPlugin CreateSingleInvalid 1`] = `
Object {
  "data": Object {
    "createDocument": null,
  },
  "errors": Array [
    Object {
      "locations": Array [
        Object {
          "column": 3,
          "line": 3,
        },
      ],
      "message": "bar should be string",
      "path": Array [
        "createDocument",
      ],
    },
  ],
}
`;

exports[`JsonValidationPlugin CreateValid 1`] = `
Object {
  "data": Object {
    "createDocument": Object {
      "document": Object {
        "id": 1,
        "meta": Object {
          "bar": "1",
          "foo": 1,
        },
      },
    },
  },
}
`;

exports[`JsonValidationPlugin UpdateInvalid 1`] = `
Object {
  "data": Object {
    "updateDocumentById": null,
  },
  "errors": Array [
    Object {
      "locations": Array [
        Object {
          "column": 3,
          "line": 3,
        },
      ],
      "message": "foo should be integer, bar should be string",
      "path": Array [
        "updateDocumentById",
      ],
    },
  ],
}
`;

exports[`JsonValidationPlugin UpdateValid 1`] = `
Object {
  "data": Object {
    "updateDocumentById": Object {
      "document": Object {
        "id": 1,
        "meta": Object {
          "bar": "2",
          "foo": 1,
        },
      },
    },
  },
}
`;
